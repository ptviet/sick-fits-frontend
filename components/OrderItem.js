import formatMoney from '../lib/formatMoney';

const OrderItem = ({ item }) => {
  return (
    <div className="order-item">
      <img src={item.image} alt={item.title} />
      <div className="item-details">
        <h2>{item.title}</h2>
        <p>Qty: {item.quantity}</p>
        <p>Each: {formatMoney(item.price)}</p>
        <p>Subtotal: {formatMoney(item.price * item.quantity)}</p>
        <p>Description: {item.description}</p>
      </div>
    </div>
  );
};

export default OrderItem;
