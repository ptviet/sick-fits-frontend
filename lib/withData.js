import withApollo from 'next-with-apollo';
import ApolloClient from 'apollo-boost';
import { endpointProd } from '../config';
import { LOCAL_STATE_QUERY } from '../components/Cart';

function createClient({ headers }) {
  return new ApolloClient({
    // uri: process.env.NODE_ENV === 'development' ? endpoint : endpointProd,
    uri: 'https://sp-sick-fits-yoga-prod.herokuapp.com',
    request: operation => {
      operation.setContext({
        fetchOptions: {
          credentials: 'include'
        },
        headers
      });
    },
    // Local data
    clientState: {
      resolvers: {
        Mutation: {
          toggleCart(_, variables, { cache }) {
            // Read the cartOpen value from the cache
            const { cartOpen } = cache.readQuery({ query: LOCAL_STATE_QUERY });
            // Write new cartOpen state
            const data = {
              data: {
                cartOpen: !cartOpen
              }
            };
            cache.writeData(data);
            return data;
          }
        }
      },
      defaults: {
        cartOpen: false
      }
    }
  });
}

export default withApollo(createClient);
