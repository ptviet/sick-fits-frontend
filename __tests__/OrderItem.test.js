import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import OrderItem from '../components/OrderItem';
import { fakeOrderItem } from '../lib/testUtils';

const item = fakeOrderItem();

describe('<OrderItem />', () => {
  it('should render and match snapshot', () => {
    const wrapper = shallow(<OrderItem item={item} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
