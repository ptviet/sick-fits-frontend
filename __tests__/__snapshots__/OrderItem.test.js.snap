// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderItem /> should render and match snapshot 1`] = `
<div
  className="order-item"
>
  <img
    alt="non omnis consequatur enim quia autem et"
    src="reprehenderit.jpg"
  />
  <div
    className="item-details"
  >
    <h2>
      non omnis consequatur enim quia autem et
    </h2>
    <p>
      Qty: 
      1
    </p>
    <p>
      Each: 
      $42.34
    </p>
    <p>
      Subtotal: 
      $42.34
    </p>
    <p>
      Description: 
      modi tenetur amet modi reprehenderit omnis adipisci
    </p>
  </div>
</div>
`;
